services/github/github-directory-file-count.service.js

```diff
@@ -0,0 +1,158 @@
+'use strict'
+
+const path = require('path')
+const Joi = require('joi')
+const { InvalidParameter } = require('..')
+const { ConditionalGithubAuthV3Service } = require('./github-auth-service')
+const {
+  documentation: commonDocumentation,
+  errorMessagesFor,
+} = require('./github-helpers')
+
+const documentation = `${commonDocumentation}
+<p>
+  <b>Note:</b><br>
+  1. Parameter <code>type</code> accepts either <code>file</code> or <code>dir</code> value. Passing any other value will result in an error.<br>
+  2. Parameter <code>extension</code> accepts file extension without a leading dot.
+     For instance for <code>.js</code> extension pass <code>js</code>.
+     Only single <code>extension</code> value can be specified.
+     <code>extension</code> is applicable for <code>type</code> <code>file</code> only.
+     Passing it either without <code>type</code> or along with <code>type</code> <code>dir</code> will result in an error.<br>
+  3. GitHub API has an upper limit of 1,000 files for a directory.
+     In case a directory contains files above the limit, a badge might present inaccurate information.<br>
```

Comment:
I think this is a reasonable limitation, especially since the `contents` API seems much like it'll be easier for us to work with than the alternative `trees` API and how uncommon a >1k path and badge scenario would be :+1: